### PR TITLE
REALMC-6683: Add a cyclical tracking field to _object

### PR DESCRIPTION
**Pull request by haleyowen**

### CODE REVIEW

File: otto.go [hunk 1]
```diff
@@ -707,6 +707,18 @@ type Object struct {
 	value  Value
 }
 
+func (self Object) IncCyclicalCount() {
+	self.object.cyclicalCount++
+}
+
+func (self Object) DecCyclicalCount() {
+	self.object.cyclicalCount--
+}
```
Comment (tsedgwick) [inline otto.go:716]: do these need a mutex?

**Comment by tsedgwick** [comment]
did the linked list not pan out?

File: otto.go (hunk 1 above)
Comment (haleyowen) [reply otto.go:716]: oh yea probably

**Comment by haleyowen** [comment]
> did the linked list not pan out?

i couldn't get it to work, and this was also just simpler

**Pull request merged**